BootloaderCorePkg/Library/MemoryAllocationLib/MemoryAllocationLib.c

```diff
@@ -54,8 +54,9 @@ InternalUpdateMemPoolBottom (
   Allocates a buffer of type EfiBootServicesData.
 
   Allocates the number bytes specified by AllocationSize of type EfiBootServicesData and returns a
-  pointer to the allocated buffer.  If AllocationSize is 0, then a valid buffer of 0 size is
-  returned.  If there is not enough memory remaining to satisfy the request, then NULL is returned.
+  pointer to the allocated buffer.  If AllocationSize is 0, LdrGlobal->MemPoolCurrTop is returned.
+  If there is not enough memory remaining to satisfy the request, InternalUpdateMemPoolTop ASSERTS 
+  and the function does not return.

```

Comment:
yes. this is the current code behavior to ASSERT even in release build. 
Do we have plan to support to build ASSERT out in release mode? If yes, the function will return.

Reply:
We have no plan to remove the ASSERT in release build for SBL.

Reply:
I think this is a good feedback. I can change the assert to deadloop.

Reply:
OK. then it is fine.